### ISSUE TITLE
[TF2] [Feature Request] Allow these specific weapons to be used by other classes

### ISSUE THREAD

**Issue by AzureWoof** [ValveSoftware/Source-1-Games#5313]
There are specific weapons in the game that wouldn't cause any real balance issues if they were to be added to the list of multi-class weapons. In fact, they would freshen up the game and give players extra options while also not being silly. These typically consist of shotguns and pistols, which are weapons who's stock variants are already available to multiple classes at once.

Scout and Engineer for example both have the ability to use the stock pistol, while the Soldier, Pyro, Heavy, and Engineer have the ability to use the stock shotgun.

Certain weapon unlocks are already available for multiple classes at once, but there's currently some missed potential for other weapon unlocks that are currently restricted to one specific class.

I'd like to request for the following weapons to be added to other classes who can make good use of them:

**Family Business** _(Currently restricted to Heavy, would benefit from being added to Soldier, Pyro, and Engineer)_
**Pretty Boy's Pocket Pistol** _(Currently restricted to Scout, would benefit from being added to Engineer)_
**Winger** _(Currently restricted to Scout, would benefit from being added to Engineer)_

**Comment by JoriKos** [comment]
The Family Business would not make any sense on the other classes, why would any of them run around with a shotgun that has a hammer and sickle on it? I also don't think the +3 HP on hit will do much for Engi considering he has a Dispenser he can retreat to, the extra jump height on Engineer is more likely to balance issues on maps because he can potentially reach more places.

**Comment by AzureWoof** [comment]
> The Family Business would not make any sense on the other classes, why would any of them run around with a shotgun that has a hammer and sickle on it?

The Reserve Shooter was added to the Pyro in the past, despite the slight design disparity when it comes to the character's personality. So I don't really see this as an issue. 

> I also don't think the +3 HP on hit will do much for Engi considering he has a Dispenser he can retreat to, 

The faster firing speed would come in handy in tight situations, and the extra health on hit would be beneficial when away from your dispenser.

> the extra jump height on Engineer is more likely to balance issues on maps because he can potentially reach more places.

The Engineer can already jump off of his buildings and sentry jump with the wrangler. I don't see how this would be a balance issue.

**Comment by JoriKos** [comment]
> The Reserve Shooter was added to the Pyro in the past, despite the slight design disparity when it comes to the character's personality. So I don't really see this as an issue.

Yeah and it was changed so airblast counted for the Mini-Crits, which was a horrible change, and now it's useless on Pyro because the actual stat design of the weapon doesn't fit well on him. Besides, the Pyro is a more ambiguous character where the Reserve Shooter can fit well on. If you gave the Reserve Shooter to Engi or Heavy it would feel weird.

> The faster firing speed would come in handy in tight situations, and the extra health on hit would be beneficial when away from your dispenser.

You'd be better off having a different unlock entirely than allowing the Engi to use the PBPP, the 3 HP isn't really gonna matter (just like it barely matters on Scout) and you're better off walking to your Dispenser instead (which the Scout can't do, because he can't build them so he doesn't know where one is or if there even is one).

> The Engineer can already jump off of his buildings and sentry jump with the wrangler. I don't see how this would be a balance issue.

The Wrangler jumps take most of his health and usually end up killing the Engi or leaving him barely alive and requires a lot of time to set up a level 3 Sentry. Jumping on buildings also doesn't guarantee you can get up on every spot, while this increase in jump height may be a way safer way of getting up on certain spots that you couldn't before. Not to mention that the Winger would look very odd on Engineer because it has the wings and atomic symbol that's associated with Scout.

It's not a bad thing to have weapons be class-exclusive, especially because many of them were designed to fit the class' aesthetic and they'd look out of place on other characters. There's probably a reason this was only done for the Reserve Shooter on Pyro.

**Comment by AzureWoof** [comment]
> Yeah and it was changed so airblast counted for the Mini-Crits, which was a horrible change, and now it's useless on Pyro because the actual stat design of the weapon doesn't fit well on him. 

While it's unfortunate the Reserve Shooter turned out that way on Pyro, that doesn't have anything to do with this request.

> Besides, the Pyro is a more ambiguous character where the Reserve Shooter can fit well on. If you gave the Reserve Shooter to Engi or Heavy it would feel weird.

That's entirely subjective.

> You'd be better off having a different unlock entirely than allowing the Engi to use the PBPP, the 3 HP isn't really gonna matter (just like it barely matters on Scout) and you're better off walking to your Dispenser instead (which the Scout can't do, because he can't build them so he doesn't know where one is or if there even is one).

Like I mentioned previously, the extra firing speed would be extremely useful, and the health on hit would be beneficial when away from your dispenser. Just because you're better off going to your dispenser doesn't mean you always can. Additionally, not everyone plays Engineer stuck next to a dispenser. The Gunslinger already makes the Engineer a lot more mobile, so that's a clear choice when it comes to a weapon that complements this change.

> The Wrangler jumps take most of his health and usually end up killing the Engi or leaving him barely alive and requires a lot of time to set up a level 3 Sentry.

Not always the case. You don't need a level 3 Sentry to Sentry jump. It's extremely easy to Sentry jump off of multiple other levels of sentries, along with the mini sentry, while saving a majority of your health, which you can easily heal back anyway.

> Jumping on buildings also doesn't guarantee you can get up on every spot, while this increase in jump height may be a way safer way of getting up on certain spots that you couldn't before.

Jumping off of buildings would already give you the ability to get to all of these spots. What I'm assuming you're concerned about is the extra jump height on top of jumping off of buildings, which would be a valid concern. However, the wrangler already allows you to get to all of these spots, and as mentioned previously, you don't need to lose too much health to do so.

> Not to mention that the Winger would look very odd on Engineer because it has the wings and atomic symbol that's associated with Scout.

I can kind of see this? But for the same reason as mentioned earlier, it wouldn't be too much of an issue.

> It's not a bad thing to have weapons be class-exclusive, especially because many of them were designed to fit the class' aesthetic and they'd look out of place on other characters. There's probably a reason this was only done for the Reserve Shooter on Pyro.

It's not a bad thing to give players more options when it comes to their playstyle. Just because you personally wouldn't use these weapons if they were added to other classes doesn't mean others wouldn't. The Family Business is probably the strongest candidate for this out of the three, and the only thing I've heard against it is that it would "look weird".

The Reserve Shooter was added to Pyro because he could easily make use of it's stats at the time, regardless of if it looked out of place visually. Gameplay should always be considered above all else, and all of these weapons would be beneficial if added to these other classes.

**Comment by gabrielwoj** [comment]
Using the Wrangler's right-click on an enemy's feet so they go airborne by the rocket explosion, then switching to the Reserve Shooter for a mini-crit sounds really cool, though. I don't think it would be very viable, though, but cool nonetheless.

**Comment by AzureWoof** [comment]
> Using the Wrangler's right-click on an enemy's feet so they go airborne by the rocket explosion, then switching to the Reserve Shooter for a mini-crit sounds really cool, though. I don't think it would be very viable, though, but cool nonetheless.

True, but that's why I didn't add the Reserve Shooter to the list, as it's upsides wouldn't really come into play on all of the other classes. Wouldn't want it to have the same fate as being on Pyro.

If it gets an entire rebalance in the future, I wouldn't mind seeing it on other classes though, as long as they can actually make use of it.

**Comment by JoriKos** [comment]
> While it's unfortunate the Reserve Shooter turned out that way on Pyro, that doesn't have anything to do with this request.

It's an example of a single-class weapon that was turned into a multi-class, adjusted for said class, but then it turned out badly. I also didn't bring up the Reserve Shooter, that was you.

> Like I mentioned previously, the extra firing speed would be extremely useful [...]

The health on hit is already pretty much completely negligible on Scout, undoubtedly it will suffer the same fate on Engineer. The only use I can maybe see is Battle Engi, but I don't think it's worth it to give Engi a pistol that doesn't suit him for a single playstyle that still doesn't gain a lot of benefit from having this pistol as an option.

> Not always the case. You don't need a level 3 Sentry to Sentry jump. It's extremely easy to Sentry jump off of multiple other levels of sentries, along with the mini sentry, while saving a majority of your health, which you can easily heal back anyway.

You still sacrifice a damaging secondary just to reach a spot, which is still a big sacrifice to make if you're gonna be playing an Engi playstyle that doesn't stay near his Sentry. It would also be incredibly unfitting to Engi to give him extra jump height, because he's not a mobile class.

> it wouldn't be too much of an issue.

I don't see how it wouldn't be, besides "I personally don't care about it", which doesn't mean it's not an issue.

> It's not a bad thing to give players more options when it comes to their playstyle. [...]

Giving classes weapons that don't fit them outside of an incredibly niche playstyle (at max) and generally won't be used is a bad idea. It just gives more weapons that you won't be using on it, I can't see any reason for any class to use the Family Business over the Panic Attack so it will just have an incredibly small usage rate.

> The Reserve Shooter was added to Pyro because he could easily make use of it's stats at the time [...]

Literally everyone could, because it had an incredibly problematic design. If you JUMPED it would give Mini-Crits, it should have never had such a design. Currently, it's a good combo weapon on Soldier, but entirely useless on Pyro. It shouldn't really ever have been given to Pyro in the first place.

> Gameplay should always be considered above all else

This is simply not true, gameplay is important, but having weapons whose visual design is specifically made for one class transferred to every other class just because "someone might want to use it" is a bad idea. A similar thing people always want is adding a Market Gardener, Gunboats, or Beggar's Bazooka equivalent to Demo. Stuff like this is more likely to just remove a unique aspect of one class than add more options for other classes.

I don't see much of a point in arguing this, none of these weapons would add anything to the other classes and their designs would clash with the new ones. You might not care about this but there are already enough cosmetics that make no sense to be multi/all-class, let's not do this with weapons too.

**Comment by JoriKos** [comment]
> Using the Wrangler's right-click on an enemy's feet so they go airborne by the rocket explosion, then switching to the Reserve Shooter for a mini-crit sounds really cool, though. I don't think it would be very viable, though, but cool nonetheless.

If you right click on an enemy's feet, it will lock on to the player and shoot them in the body instead, which means it's even more rare to actually get a Reserve Shooter hit as Engi.

**Comment by AzureWoof** [comment]
> It's an example of a single-class weapon that was turned into a multi-class, adjusted for said class, but then it turned out badly. I also didn't bring up the Reserve Shooter, that was you.

You brought up the fact that it turned out badly, which wasn't part of the discussion. Just because that weapon turned out badly doesn't mean these ones will if this change was added.

> The health on hit is already pretty much completely negligible on Scout, undoubtedly it will suffer the same fate on Engineer. 

I disagree. Myself and many others have made plenty of good use out of it in the past. And like I previously mentioned, the faster firing speed does come into play, which it seems like you keep omitting for some reason.

> The only use I can maybe see is Battle Engi, but I don't think it's worth it to give Engi a pistol that doesn't suit him for a single playstyle that still doesn't gain a lot of benefit from having this pistol as an option.

It will always come into play whenever you're not next to your dispenser. There can be multiple reasons for finding yourself in that situation, and it doesn't only benefit that specific playstyle. Additionally, saying it doesn't suit him is entirely subjective, especially considering that nothing visually about the weapon is unfit for Engineer, which seems to be your main argument here for whatever reason.

> You still sacrifice a damaging secondary just to reach a spot, which is still a big sacrifice to make if you're gonna be playing an Engi playstyle that doesn't stay near his Sentry.

The jump height increase of the Winger is negligible compared the the height and distance you get from the Wrangler.

> It would also be incredibly unfitting to Engi to give him extra jump height, because he's not a mobile class.

Entirely subjective. Additionally, he can be a mobile class with the Wrangler. It could be said that giving him extra minor mobility options wouldn't be unfitting with that in mind.

> I don't see how it wouldn't be (too much of an issue), besides "I personally don't care about it", which doesn't mean it's not an issue.

Don't put words in my mouth. I said it wouldn't be too much of an issue because balance-wise, these would be extremely minor changes at best that get completely outshined by other weapons in their arsenals. You have yet to make one solid argument as to how any of these changes would be significantly damaging in any way to gameplay aside from describing use cases that can already be done by other weapons in the game.

> Giving classes weapons that don't fit them outside of an incredibly niche playstyle (at max) and generally won't be used is a bad idea. It just gives more weapons that you won't be using on it,

Like I mentioned previously, just because you wouldn't personally use them doesn't mean others wouldn't. These additions wouldn't be stuck being used in "niche" playstyles at all. You yourself attested to this by mentioning your concern about the Winger being used to reach higher places without the use of a Wrangler, which can be done in any kind of playstyle and is hardly niche. The Pretty Boy's Pocket Pistol's upsides, as mentioned previously, also comes into play extremely often. Especially the firing speed increase.

> I can't see any reason for any class to use the Family Business over the Panic Attack so it will just have an incredibly small usage rate.

The Family Business is an extremely well designed weapon stuck on a class that cannot make use of it very often. The Heavy already has his Miniguns, and people tend to pick his lunchbox items as secondaries over the Shotgun. Giving this weapon to other classes would immediately increase it's usage rate. The Panic Attack has it's specific usage, as does the Family Business. The extra firing speed and clip size would come into play way a lot more often on those other classes, since they tend to use their shotguns way more often.


> > The Reserve Shooter was added to Pyro because he could easily make use of it's stats at the time [...]
> 
> Literally everyone could, because it had an incredibly problematic design. If you JUMPED it would give Mini-Crits, it should have never had such a design. Currently, it's a good combo weapon on Soldier, but entirely useless on Pyro. It shouldn't really ever have been given to Pyro in the first place.

That weapon design at the time was absolutely broken when added to Pyro. Everyone agrees with that fact. However, none of the weapons I mentioned in this post would have this kind of effect. So, it's a moot point.

> > Gameplay should always be considered above all else
> 
> This is simply not true, gameplay is important, but having weapons whose visual design is specifically made for one class transferred to every other class just because "someone might want to use it" is a bad idea.

It hardly affects anything visually. Nobody is going to mistake the Engineer for the Scout because he's using the Winger. You probably won't be focusing on this at all while playing the game.

> A similar thing people always want is adding a Market Gardener, Gunboats, or Beggar's Bazooka equivalent to Demo.

Not comparable.

> Stuff like this is more likely to just remove a unique aspect of one class than add more options for other classes.

I'd rather a weapon be more useful than to keep it stuck as a "unique aspect of one class" that's stuck gathering dust in someone's inventory because it's completely overshadowed by other weapons of the class it's designed for.

> I don't see much of a point in arguing this, none of these weapons would add anything to the other classes

I've already discussed why this isn't the case.

> and their designs would clash with the new ones.

Not really.

> You might not care about this but there are already enough cosmetics that make no sense to be multi/all-class, let's not do this with weapons too.

They might not make any sense to you, but there's absolutely no harm in cosmetics being multi/all-class at all. If you don't like it, just don't wear it. Simple as that. Same thing can be said for these weapons.

**Comment by AzureWoof** [comment]
Sending people to negatively react to my posts instead of making relevant points doesn't help your argument, by the way. If you have a genuine reason as to why this is a bad idea, please elaborate.

**Comment by JoriKos** [comment]
> Sending people to negatively react to my posts instead of making relevant points doesn't help your argument, by the way.

I didn't, people just disagree with your post. I didn't send anyone anywhere, assuming I did just because people don't agree with your opinion is not respectful either. Assume good faith. I did bring my genuine reasons as to why I think it is a bad idea, and I have elaborated, but it's not a very "respectful discussion" if you assume my reasons are not genuine and that I ask people to come and leave a thumbs down on your post.

**Comment by AzureWoof** [comment]
> I didn't, people just disagree with your post. I didn't send anyone anywhere, assuming I did just because people don't agree with your opinion is not respectful either. Assume good faith. I did bring my genuine reasons as to why I think it is a bad idea, and I have elaborated, but it's not a very "respectful discussion" if you assume my reasons are not genuine and that I ask people to come and leave a thumbs down on your post.

I've been taking a close look at this thread and it's very obviously apparent that you're doing that. It would have to be pretty convenient timing for other people to immediately react the post right after you applied your reaction. Regardless, it doesn't matter. My point still stands.

**Comment by JoriKos** [comment]
> I've been taking a close look at this thread and it's very obviously apparent that you're doing that.

There were already like 3 reactions to your original post before I even left more than 1 comment. I don't care about this discussion as much as you think I do, it's not that important. If anything, Gabriel would not support your argument because they are one of the 2 people that have somehow participated in this discussion that I actually know.

**Comment by AzureWoof** [comment]
> > I've been taking a close look at this thread and it's very obviously apparent that you're doing that.
> 
> There were already like 3 reactions to your original post before I even left more than 1 comment. I don't care about this discussion as much as you think I do, it's not that important. If anything, Gabriel would not support your argument because they are one of the 2 people that have somehow participated in this discussion that I actually know.

It's not those reactions that I'm talking about. My previous reply was reacted to in rapid succession. I stopped assuming good faith when you put words in my mouth that I never said to further your argument earlier. Regardless, I'm not interested in arguing about this any further. It's such a silly thing to do.

**Comment by JoriKos** [comment]
> My previous reply was reacted to in rapid succession.

Probably because people might subscribe to a semi-active thread considering they're not super common on this GitHub. I have nothing against you or you trying to bring a discussion with this idea, the thumbs down (as far as I know) just show you disagree with the idea. I didn't wanna argue further about it, so I just left it and then didn't check the thread until I got a notif for a new comment.

**Comment by Platina6978** [comment]
> > My previous reply was reacted to in rapid succession.
> 
> Probably because people might subscribe to a semi-active thread considering they're not super common on this GitHub. I have nothing against you or you trying to bring a discussion with this idea, the thumbs down (as far as I know) just show you disagree with the idea. I didn't wanna argue further about it, so I just left it and then didn't check the thread until I got a notif for a new comment.

Yea, pretty much this. I tend to keep an eye on this github thread out of curiousity and out of my passion for TF2, especially more active issues. I pretty much agree with most of what JoriKos has said and disagree with the original proposal, hence the thumbs up/down. 

Other than putting more emphasis on not sacrificing any merc's personality (family business was given to heavy as a set, winger is clearly visually designed to fit scout) on top of rather seeing new weapons for these classes instead of making more things all-class, I had not much to add to JoriKos arguments. I'm stepping in for a second however because I was not sent here by anyone to react to anything in any particular way. I just personally disagree with the initial idea here, that's all.
